### PR TITLE
Closes #1589 created a new function to convert NA to blank and added …

### LINKED ISSUE #1589: New function: convert_na_to_blanks()
See discussion at https://github.com/atorus-research/xportr/issues/63

We need a reverse function to `convert_blanks_to_na()` (https://pharmaverse.github.io/admiral/reference/convert_blanks_to_na.html) that now converts NA values back to blanks.

### PR DESCRIPTION
**Pull request by sadchla-codes**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiraldev/main/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/main/articles/unit_test_guidance.html#writing-unit-tests-in-admiral-)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/main/articles/programming_strategy.html#deprecation-1)?
- [x] Update to all relevant roxygen headers and examples 
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately 
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -42,6 +42,8 @@ based on [Common Terminology Criteria for Adverse Events (CTCAE) v5.0](https://c
 
 - Removed R Version 3.6 check in CI/CD workflows in favor of the three most recent versions: 4.0, 4.1 and 4.2. (#1556)
 
+- New function `convert_na_to_blanks()` converts `NA` strings to blank while preserving all attributes
```
Comment (bundfussr) [inline NEWS.md:45]: Could you add the issue number?

File: R/admiral-package.R [hunk 2]
```diff
@@ -4,7 +4,7 @@
 #' @importFrom dplyr arrange bind_rows case_when desc ends_with filter full_join group_by
 #'             if_else mutate mutate_at mutate_if n pull rename rename_at row_number select slice
 #'             semi_join starts_with transmute ungroup vars n_distinct union distinct
-#'             summarise_at summarise coalesce bind_cols na_if tibble
+#'             summarise_at summarise coalesce bind_cols na_if tibble across everything
```
Comment (bundfussr) [inline R/admiral-package.R:7]: Could you add the new functions such that they are ordered alphabetically?

File: tests/testthat/test-user_utils.R [hunk 3]
```diff
@@ -1,13 +1,15 @@
 # convert_blanks_to_na ----
 ## Test 1: blank strings are turned into `NA` ----
-test_that("convert_blanks_to_na Test 1: blank strings are turned into `NA`", {
+# ---- user_utils, test 1: convert_blanks_to_na Test 1: blank strings are turned into `NA` ----
```
Comment (bundfussr) [inline tests/testthat/test-user_utils.R:3]: It seems something went wrong here. Could you fix it? Usually calling the "Format test_that test file" addin should produce the correct comments and descriptions.

File: R/user_utils.R [hunk 4]
```diff
@@ -87,6 +87,74 @@ convert_blanks_to_na.data.frame <- function(x) { # nolint
...
+#'
+#' df <- tibble(
+#'   a = structure(c("a", "b", NA, "c"), label = "A"),
+#'   b = structure(c(1, NA, 21, 9), label = "B"),
+#'   c = structure(c(TRUE, FALSE, TRUE, TRUE), label = "C"),
+#'   d = structure(c(NA, NA, "s", "q"), label = "D")
+#' )
+#' print(df)
+#' convert_na_to_blanks(df)
+convert_na_to_blanks <- function(x) {
```
Comment (bundfussr) [inline R/user_utils.R:126]: Could you implement `convert_na_to_blanks()` as S3 generic like `convert_blanks_to na()`?

I.e., the following calls should work.
```
convert_na_to_blanks(list(NA_character_, "1"))
convert_na_to_blanks(tibble(AVALC = c("1", NA_character_)))
```

For this you need to update `convert_na_to_blanks()` such that `UseMethod("convert_na_to_blanks")` is called and rename the other functions such that the form `<method name>.<class name>()` is used, e.g., `convert_na_to_blanks.default()` or `convert_na_to_blanks.data.frame()`.

See https://adv-r.hadley.nz/s3.html#s3-methods for details on S3 generics.

**Review by bundfussr — CHANGES_REQUESTED** [review]

File: tests/testthat/test-user_utils.R (hunk 3 above)
Comment (sadchla-codes) [reply tests/testthat/test-user_utils.R:3]: these comments and descriptions were produced using the **Format test_that test file** and I have to fix the issue above but missed that one. I noticed there are a few more test programs that aren't formatted using the agreed format so I will will **Format test_that test file** to make sure there isn't a bug.

**Review by bundfussr — CHANGES_REQUESTED** [review]

**Review by bundfussr — APPROVED** [review]

**Review by bms63 — APPROVED** [review]
LGTM!

**Pull request merged**